code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm

```diff
@@ -7,6 +7,26 @@
 	unacidable = 1 //glass
 	center_of_mass = list("x"=16, "y"=10)
 
+/obj/item/weapon/reagent_containers/food/drinks/standard_feed_mob(var/mob/user, var/mob/target)
+	// Special snowflake code inbound.
+	if(is_open_container() && user == target && user.a_intent != I_HELP && user.a_intent != I_HURT) // This is a pretty specific case.
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			if(!H.check_has_mouth())
+				user << "Where do you intend to put \the [src]? You don't have a mouth!"
+				return
+			var/obj/item/blocked = H.check_mouth_coverage()
+			if(blocked)
+				user << SPAN_WARNING("\The [blocked] is in the way!")
+				return
+
+		user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN) //puts a limit on how fast people can eat/drink things
+		user.visible_message(SPAN_NOTICE("[user] gulped down the whole [src]!"),SPAN_NOTICE("You gulped down the whole [src]!"))
```

Comment:
This needs a do_after of about 5-8 seconds

and if interrupted the whole drink should spill all over the user's turf

Reply:
Will do.

Reply:
Right, not a preset time, base it on the quantity of reagents

do_after(user, reagents.total_volume)

big bottles will take a lot, small quantites will take almost nothing